templates/repo/empty.tmpl

```diff
@@ -50,14 +50,17 @@ git push -u origin {{.Repository.DefaultBranch}}</code></pre>
 git push -u origin {{.Repository.DefaultBranch}}</code></pre>
 								</div>
 							</div>
-							<script defer>
-								/* eslint-disable no-undef */
-								const cloneUrls = document.getElementsByClassName('clone-url');
-								if (cloneUrls) {
-									for (let i = 0; i < cloneUrls.length; i++) {
-										cloneUrls[i].textContent = (isSSH ? sshButton : httpsButton).getAttribute('data-link');
+							<script>
+								(() => {
+									const proto = localStorage.getItem('repo-clone-protocol') || 'https';
+									const btn = document.getElementById(`repo-clone-${proto}`) || document.getElementById(`repo-clone-https`) || document.getElementById(`repo-clone-ssh`);
+									if (btn) {
+										const cloneUrls = document.getElementsByClassName('clone-url');
+										for (let i = 0; i < cloneUrls.length; i++) {
+											cloneUrls[i].textContent = btn.getAttribute('data-link');
+										}
 									}
-								}
+								})();
```

Comment:
Why the IIFE?

Reply:
Why to pollute the global namescape ....

Reply:
`const something` does not pollute `window`, I'm not sure what you mean.

Reply:
Because the old code was using global variables, and that's how bug comes ... when another template is changed, then here the bug occurs. I just want to make sure that there would be no more such bugs.

If you have strong against to the IIFE, I can remove it.

Reply:
That's the code in another template, can you accept such code style?


![image](https://user-images.githubusercontent.com/2114189/160097687-bc02e8e8-bfa9-469a-a77e-2480e5ff50d0.png)

Reply:
I think we should not pointlessly add IIFEs.

Also, why is above code disabling eslint? What rule is triggering?

Reply:
To my knowledge each `<script>` tag creates its own JS scope (function scope), so there should not be any leakage of `const` variables to the outside world.

Reply:
Also, if the aim is just to create scope, you can wrap code in `{}` to create a [block scope](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/block#block_scoping_rules_with_let_const_or_function_declaration_in_strict_mode) which `let` and `const` will honour, no need for the extra overhead of an IIFE.

Reply:
Also, we should probably put all inline scripts into strict mode via `'use strict';` or `type=module`.

Reply:
Then, how the old code works? Where does these `isSSH` come from?

Does the old template https://github.com/go-gitea/gitea/blame/ea46142bcea6084e50a11baf048f8eb8862285c7/templates/repo/clone_buttons.tmpl  leaks the `isSSH`/`sshButton`/`httpsButton` to here?

Reply:
And this is really a simple question, you can try it by yourself:

```
<script>
const a=1;
</script>

<script>
alert(a);
</script>
```

Reply:
> Also, we should probably put all inline scripts into strict mode via `'use strict';` or `type=module`.

You are more than welcome to propose a perfect PR, I am not expert.

Reply:
You're right, it seems script tags share a scope, didn't know about this "feature".

We can enforce separate scope for each script with `type=module`, but it can be done in another PR. Example https://jsfiddle.net/silverwind/cmd18jtf/1/ throws `ReferenceError: a is not defined`.